client/src/components/Chat/MessageCreator.js

```diff
@@ -57,15 +59,35 @@ const useStyles = makeStyles(theme => ({
     },
 }));
 
-const MessageCreator = () => {
+const MessageCreator = ({groupId}) => {
     const classes = useStyles();
+    const { profile } = useGlobalContext();
+    const { user } = profile;
+    const { conversationManager } = useConversationContext();
+    const [workingText, setWorkingText] = React.useState('');
+
+    const sendMessage = () => {
+        const payload = {
+            user: user._id, 
+            text: workingText, 
+            image: profile.imageUrl
```

Comment:
Because we store the pic in the message, if the user changes their profile pic, the older messages will still have the old one. which is kinda weird. it would be better if the frontend would fetch all profile images for a group. then in each message you would do something like:
```jsx
<img src={profileImages[userId]}> 
```

Reply:
That being said, this is not a priority, since your implementation is working fine otherwise. But if there's time before the demo, you can try to address this

Reply:
I changed the Conversation model to use the users profileId, updated the GET conversations route to populate the messages with the profile image and userId, and updated the socket message handler to return that same type of message (populated with profile fields). So the profile picture will get updated in the Chat whenever the user refreshes the page. Benefit to this is we don't have to make a call to get all of the chat members profiles every time we go to the chat page, the drawback is that if the user updates their profile picture historic messages won't be updated for other group members until they either logout or refresh the page.

Reply:
It is still kind of weird, but now it's a weirdness that fixes itself eventually haha. If necessary I could store a dictionary in the Chat page mapping the members of the chat to the imageurl and set the imageurl of the group member to that of the incoming message to make sure its updated. Then pass that dictionary down to use as the avatar image. I think that would fix it?

Reply:
I think hat would do. But what you did is good for now!